apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template-inserter.php

```diff
@@ -0,0 +1,261 @@
+<?php
+/**
+ * Full site editing file.
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE;
+
+/**
+ * Class WP_Template_Inserter
+ */
+class WP_Template_Inserter {
+	/**
+	 * Modern Business header content.
+	 *
+	 * @var string $header_content
+	 */
+	private $header_content;
+
+	/**
+	 * Modern Business footer content.
+	 *
+	 * @var string $footer_content
+	 */
+	private $footer_content;
+
+	/**
+	 * Current theme slug.
+	 *
+	 * @var string $theme_slug
+	 */
+	private $theme_slug;
+
+	/**
+	 * This site option will be used to indicate that template data has already been
+	 * inserted for this theme, in order to prevent this functionality from running
+	 * more than once.
+	 *
+	 * @var string $fse_template_data_option
+	 */
+	private $fse_template_data_option;
+
+	/**
+	 * WP_Template_Inserter constructor.
+	 *
+	 * @param string $theme_slug Current theme slug.
+	 */
+	public function __construct( $theme_slug ) {
+		$this->theme_slug     = $theme_slug;
+		$this->header_content = '';
+		$this->footer_content = '';
+
+		/*
+		 * Previously the option suffix was '-fse-template-data'. Bumping this to '-fse-template-data-v1'
+		 * to differentiate it from the old data that was not provided by the API. Note that we don't want
+		 * to tie this to plugin version constant, because that would trigger the insertion on each plugin
+		 * update, even when it's not necessary (it would duplicate existing data).
+		 */
+		$this->fse_template_data_option = $this->theme_slug . '-fse-template-data-v1';
```

Comment:
Is the versioning really needed?
Right now it's only useful when testing this PR for the first time (then, we will still need to delete the `-v1` option), and I guess it might cause trouble down the line. 🤔

Reply:
I think it's preferable aside from testing this PR. For example, on Dotcom we might want to clean up the existing data that Modern Business themes inserted, that's now out of date. The only way to differentiate it on those sites is by the theme option. And if we mark new data with it we'll lose that ability. It's also beneficial if we want to force template data updates in the future (for emergency fixes let's say), see https://github.com/Automattic/wp-calypso/pull/35190#discussion_r311619147.

Reply:
+1 for comparing versions or numbers vs inspecting content. 

Nonblocking question: do we also need a bool to check if content was modified by the user? In most cases if we did update template content, I wouldn't want to override any customer updates. Only if they have defaults.

Reply:
> Nonblocking question: do we also need a bool to check if content was modified by the user? In most cases if we did update template content, I wouldn't want to override any customer updates. Only if they have defaults.

I was thinking about this before, and yes, I think it would make sense to add it. Although I'm not sure about exact implementation details it should be possible. We might want to consider granularity to - e.g. do we just want to know that "some template has been updated" vs "these template IDs have been updated".